src/NServiceBus.Core/Pipeline/Outgoing/DispatchContext.cs

```diff
@@ -0,0 +1,31 @@
+namespace NServiceBus.TransportDispatch
+{
+    using Routing;
+    using Pipeline;
+    using Transports;
+
+    /// <summary>
+    /// Context for the dispatch part of the pipeline.
+    /// </summary>
+    public class DispatchContext : BehaviorContext
```

Comment:
How about renaming it to `RoutingContext`? With the immediate one, this guy does not do any dispatch. And then we can rename the immediate to just `DispatchContext` because immediate does not make sense in the context of the batched pipeline (`BatchToImmediateDispatchConnector` would become just `BatchToDispatchConnector`)